mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMailboxManager.java

```diff
@@ -442,7 +446,7 @@ public Mailbox deleteMailbox(MailboxId mailboxId, MailboxSession session) throws
             if (mailbox == null) {
                 throw new MailboxNotFoundException(mailboxId);
             }
-            assertIsOwner(session, mailbox.generateAssociatedPath());
+            assertSessionHasAdministrativeAccessOn(session, mailbox.generateAssociatedPath());
```

Comment:
Before we were throwing a "MailboxNotFoundException" in order not to leak that the mailbox exists.

Now, we are returning an InsufficientRightsException. 

This was done on purpose. I won't be surprised that some 'delegation' JMAP integration test breaks.

Can you find a way to still return the right exception here and above (by passing an exception supplier to assertSessionHasAdministrativeAccessOn ?)? Can we adda test in MbxManager for this?

Reply:
Hm went a bit fast here indeed... I guess it's not necessary to add tests for that, just modify the ones concerned in MailboxManagerTest, like instead of always check that the exception thrown is of `MailboxException` instance, directly check those two more refined ones

Reply:
Yes

Reply:
Actually took care of it regarding the tests in the other PR https://github.com/linagora/james-project/pull/3041/commits/5afbebec357b55d751dd2d45e1b2e731c734924b